src/data/path.ts

```diff
@@ -0,0 +1,206 @@
+import { isValidStringProperty } from '../validation/string';
+import { DbVersionSupport } from '../utils/dbVersion';
+
+const objectsPathPrefix = '/objects';
+
+export class ObjectsPath {
+  private dbVersionSupport: DbVersionSupport;
+
+  constructor(dbVersionSupport: DbVersionSupport) {
+    this.dbVersionSupport = dbVersionSupport;
+  }
+
+  buildCreate(consistencyLevel: string | undefined): Promise<string> {
+    return this.build({ consistencyLevel }, [this.addQueryParams]);
+  }
+  buildDelete(
+    id: string,
+    className: string,
+    consistencyLevel: string | undefined
+  ): Promise<string> {
+    return this.build({ id, className, consistencyLevel }, [
+      this.addClassNameDeprecatedNotSupportedCheck,
+      this.addId,
+      this.addQueryParams,
+    ]);
+  }
+  buildCheck(id: string, className: string): Promise<string> {
+    return this.build({ id, className }, [
+      this.addClassNameDeprecatedNotSupportedCheck,
+      this.addId,
+    ]);
+  }
+  buildGetOne(
+    id: string,
+    className: string,
+    additionals: any,
+    consistencyLevel: string | undefined,
+    nodeName: any
+  ): Promise<string> {
+    return this.build(
+      { id, className, additionals, consistencyLevel, nodeName },
+      [
+        this.addClassNameDeprecatedNotSupportedCheck,
+        this.addId,
+        this.addQueryParams,
+      ]
+    );
+  }
+  buildGet(
+    className: string | undefined,
+    limit: any,
+    additionals: any,
+    after: string
+  ): Promise<string> {
+    return this.build({ className, limit, additionals, after }, [
+      this.addQueryParamsForGet,
+    ]);
+  }
+  buildUpdate(
+    id: string,
+    className: string,
+    consistencyLevel: string | undefined
+  ): Promise<string> {
+    return this.build({ id, className, consistencyLevel }, [
+      this.addClassNameDeprecatedCheck,
+      this.addId,
+      this.addQueryParams,
+    ]);
+  }
+  buildMerge(
+    id: string,
+    className: string,
+    consistencyLevel: string | undefined
+  ): Promise<string> {
+    return this.build({ id, className, consistencyLevel }, [
+      this.addClassNameDeprecatedCheck,
+      this.addId,
+      this.addQueryParams,
+    ]);
+  }
+
+  build(params: any, modifiers: any): Promise<string> {
```

Comment:
Curious about the inconsistency of newlines in between this class' methods.

For example there is not a newline in between all the above methods up to the constructor, but starting here at `build` there is one, and below there is no newline between the remaining methods

Reply:
I don't think there's a linting rule enforcing this currently, but we should be able to add it. Should I?